DependencyInjection/Compiler/DescriptionEnhancerPass.php

```diff
@@ -59,12 +70,26 @@ public function process(ContainerBuilder $container)
             ));
         }
 
+        // Remove inactive enhancers from the container.
+        //
+        // Note that previously we relied on the container automatically removing
+        // unreferenced definitions, but this proved to not always be the case. It
+        // is safer to remove them here.
         $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
```

Comment:
I encountered the problem from the Twig service when sonata was not
present. I will have a closer look into why the Sonata enhancer was
being invoked.

On Sat, Jul 16, 2016 at 02:40:07AM -0700, Wouter J wrote:
>    In [1]DependencyInjection/Compiler/DescriptionEnhancerPass.php:
> 
>  > @@ -59,12 +70,26 @@ public function process(ContainerBuilder $container)
>  >              ));
>  >          }
>  >
>  > +        // Remove inactive enhancers from the container.
>  > +        //
>  > +        // Note that previously we relied on the container automatically removing
>  > +        // unreferenced definitions, but this proved to not always be the case. It
>  > +        // is safer to remove them here.
>  >          $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
> 
>    is there a problem when an unused service is in the cache? I wouldn't
>    think so.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly, [2]view it on GitHub, or [3]mute the thread.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/resource-bundle/pull/34#discussion_r71063863
>    2. https://github.com/symfony-cmf/resource-bundle/pull/34/files/951c5b077f208c5f302c26bf170568fc73d07d56#r71063863
>    3. https://github.com/notifications/unsubscribe-auth/AAgZcZVAA89o1LsLJ26MRsqdvErs4bz_ks5qWKb3gaJpZM4I6bpJ
>    4. https://github.com/symfony-cmf/resource-bundle/pull/34/files/951c5b077f208c5f302c26bf170568fc73d07d56#r71063863